### ISSUE TITLE
paddle2.2的paddle.vision.dataset.CIFAR100的API使用异常

### ISSUE THREAD

**Issue by Stave604671** [PaddlePaddle/Paddle#39973]
Paddle version: 2.2.0
Paddle With CUDA: True
OS: Windows 10
Python version: 3.8.10
CUDA version: 10.1.243
cuDNN version: 7.6.5
Nvidia driver version: 462.30
****************************************
CIFAR100API使用的时候，如果在一个代码里面同时加载其中的train_loader 和test_loader, test_loader能够吐出的data的shape出了问题，最大值只能吐出16，直接运行末尾附件的txt即可
from paddle.vision.datasets import Cifar100
import paddle
import paddle.vision.transforms as T
eval_transforms = T.Compose([
    T.Resize(256, interpolation='bicubic'),
    T.CenterCrop(224),
    T.ToTensor(),
    T.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225])
])
cifar100_train = Cifar100(mode='train', transform=eval_transforms)
cifar100_eval = Cifar100(mode='test', transform=eval_transforms)
train_loader = paddle.io.DataLoader(cifar100_train, batch_size=32, shuffle=True)
val_loader = paddle.io.DataLoader(cifar100_eval, batch_size=32, shuffle=False)
for i, data_train in enumerate(train_loader()):
    label_train = paddle.reshape(data_train[1], shape=[32
for i, data_eval in enumerate(val_loader()):
    data_eval = data_eval[1]
    label_eval = paddle.reshape(data_eval, shape=[32, 1])
    break
![屏幕截图 2022-02-28 003655](https://user-images.githubusercontent.com/59908449/155891084-5ef77ec8-9b0c-4382-9ac0-9f689cc34a98.jpg)
, 1])
![屏幕截图 2022-02-28 003720](https://user-images.githubusercontent.com/59908449/155891089-f48bc83b-2db3-49c2-8f85-b912720846ff.jpg)

[test.txt](https://github.com/PaddlePaddle/Paddle/files/8149005/test.txt)

**Comment by heavengate** [comment]
应该是数据集sample数不能被32整除，最后一个iter的sample数不足32，可以在DataLoader里设置drop_last=True
